_posts/2018-03-28-building-coreboot-on-minnowboard.md

```diff
@@ -0,0 +1,244 @@
+
+```sh
+docker run -u root -it -v $PWD:/home/coreboot/coreboot --rm  \
+    coreboot/coreboot-sdk:1.50 /bin/bash
+```
+(remember that it works as root, so you may want to change owner of new files
+after build.
+
+Configuration
+-------------
+
+coreboot is configured using `make menuconfig` similar to Linux Kernel (needs
+ncurses library). First in `Mainboard` menu we set platform:
+
+```
+Mainboard vendor (Intel)  --->
+Mainboard model (Minnow Max)  --->
+```
+
+Except of this we have to configure microcode and enable FSP in `Chipset` menu:
+
+```
+Include CPU microcode in CBFS (Include external microcode header files)  --->
+(M0130673321.h  M0130679901.h  M023067221D.h) List of space separated microcode header
+
+[*] Use Intel Firmware Support Package
+(../intel/fsp/baytrail/BAYTRAIL_FSP.fd) Intel FSP binary path and filename (NEW)
+```
+
+If we want to use VgaBIOS, we can configure it in `Devices`:
+
+```
+ [*] Add a VGA BIOS image
+(../intel/cpu/baytrail/vbios/Vga.dat) VGA BIOS path and filename (NEW)
+```
+
+On x86 by default, coreboot chooses `SeaBIOS` as a payload so that it provides
+legacy BIOS interface. You may want to use one of other options: GRUB (well
+known, robust Open Source boot loader), Tianocore (UEFI), etc.
+
+This is minimal configuration for MinnowBoard. However you probably want
+to decrease log level (defualt 8:SPEW is much too verbose) in `Console`
+menu:
+
+```
+Default console log level (5: NOTICE)  --->
+```
+
+Build & Flash
+-------------
+
+When configuration is complete, we can save it and run:
+
+```
+make
+```
+
+If everything goes well, output should end like this:
+
+```
+    CBFSPRINT  coreboot.rom
+
+Name                           Offset     Type           Size   Comp
+cbfs master header             0x0        cbfs header        32 none
+fallback/romstage              0x80       stage           30204 none
+cpu_microcode_blob.bin         0x7700     microcode           0 none
+fallback/ramstage              0x7780     stage           58614 none
+config                         0x15cc0    raw               118 none
+revision                       0x15d80    raw               582 none
+cmos_layout.bin                0x16000    cmos_layout      1208 none
+fallback/dsdt.aml              0x16500    raw             12528 none
+fallback/payload               0x19640    payload         68158 none
+payload_config                 0x2a0c0    raw              1593 none
+payload_revision               0x2a740    raw               239 none
+(empty)                        0x2a880    null          2969432 none
+bootblock                      0x2ff800   bootblock        1720 none
+    HOSTCC     cbfstool/ifwitool.o
+    HOSTCC     cbfstool/ifwitool (link)
+
+Built intel/minnowmax (Minnow Max)
+```
+
+So we can flash it. If you are using docker image, remember that it doesn't
+contain `flashrom` so you have to do that outside container:
+
+```
+sudo flashrom -l mb.layout -ibios -pdediprog -w build/coreboot.rom
+```
```

Comment:
@BartekLew it would be great to summarize this blog post, tell the reader that we are open to helping if problem faced as well as we can provide flashing services for those who can't or don't want to do it by themselves.

Reply:
:+1: